### PR TITLE
Gg 30 implement a session management system

### PR DESCRIPTION
**Pull request by samwiseGhandi**

### CODE REVIEW

File: cors-server/routes/authentication.ts [hunk 1]
```diff
@@ -11,8 +11,104 @@ export const encryptPassword = async function (password: string) {
...
     return success;
 };
 
-export default {encryptPassword, validateUser};
+export const findUser = async function (email, db) {
+    try {
+        const result = db
+            .prepare("SELECT id, email, username, password FROM user WHERE email = ?")
+            .get(email);
+        console.log(result.id, result.username, result.email);
```
Comment (M0ller) [inline cors-server/routes/authentication.ts]: remove console.logs?

File: cors-server/routes/authentication.ts [hunk 2]
```diff
@@ -11,8 +11,104 @@ export const encryptPassword = async function (password: string) {
...
+
+
+export const signIn = async function (server, db: any, newLogin: boolean) {
+    if (newLogin) {
+        server.post("/data/login", async (req: Request, res: Response) => {
+            // Extract the user's credentials from the request body
+            const { email, password } = req.body;
+            // Verify the user's credentials against the database
+            try {
+                const user = await findUser(email, db);
```
Comment (M0ller) [inline cors-server/routes/authentication.ts:44]: can write it like this: findUser(req.body.email, db)
so you dont need line 44

File: cors-server/routes/authentication.ts [hunk 3]
```diff
@@ -11,8 +11,104 @@ export const encryptPassword = async function (password: string) {
...
+                console.log("r49",user);
+                
+                if (!user) {
+                    return res
+                        .status(401)
+                        .json({ error: "r54 Invalid username or password.." });
+                }
+                console.log("r56", user);
+
+                const isValid = await validateUser(password, user.password);
```
Comment (M0ller) [inline cors-server/routes/authentication.ts:52]: validateUser(req.body.password, user.password);

File: cors-server/routes/authentication.ts [hunk 4]
```diff
@@ -11,8 +11,104 @@ export const encryptPassword = async function (password: string) {
 };
 
 export const validateUser = async function (password: string, hash: string) {
+    console.log("r14 password ", password," hash ", hash);
```
Comment (Nirodie) [inline cors-server/routes/authentication.ts]: Testing console log should be removed

File: src/components/Login.tsx [hunk 5]
```diff
@@ -60,7 +61,7 @@ export default function Login() {
             </div>
             <div className="text-sm">
                 <a
-                  href="/register"
+                  href="#"
```
Comment (M0ller) [inline src/components/Login.tsx:64]: can import Link from react-router-dom and use: 
<Link to="/register"
className="font-medium text-indigo-600 hover:text-indigo-500" > 
</Link>

Its more common i think

File: cors-server/routes/authentication.ts [hunk 6]
```diff
@@ -11,8 +11,104 @@ export const encryptPassword = async function (password: string) {
...
+
+export const signIn = async function (server, db: any, newLogin: boolean) {
+    if (newLogin) {
+        server.post("/data/login", async (req: Request, res: Response) => {
+            // Extract the user's credentials from the request body
+            const { email, password } = req.body;
+            // Verify the user's credentials against the database
+            try {
+                const user = await findUser(email, db);
+                console.log("r49",user);
```
Comment (Nirodie) [inline cors-server/routes/authentication.ts]: Testing console log should be removed

File: cors-server/routes/authentication.ts [hunk 7]
```diff
@@ -11,8 +11,104 @@ export const encryptPassword = async function (password: string) {
...
+                console.log("r56", user);
+
+                const isValid = await validateUser(password, user.password);
+                
+                if (!isValid) {
+                    return res
+                        .status(401)
+                        .json({ error: "Invalid username or password.." });
+                }
+                console.log("r66", user)
```
Comment (Nirodie) [inline cors-server/routes/authentication.ts]: Testing console log should be removed

File: cors-server/routes/authentication.ts [hunk 8]
```diff
@@ -11,8 +11,104 @@ export const encryptPassword = async function (password: string) {
...
+                // Create a new JWT with the user's information
+                const payload = { id: user.id };
+                const options = {
+                    expiresIn: "2h",
+                };
+                const secret_key = "secret_key"
+                
+                const token = jwt.sign(payload, secret_key, options);
+                console.log("r73 token ", token);
```
Comment (Nirodie) [inline cors-server/routes/authentication.ts]: Testing console log should be removed

File: cors-server/routes/authentication.ts [hunk 9]
```diff
@@ -11,8 +11,104 @@ export const encryptPassword = async function (password: string) {
...
+
+                // Store the JWT in the user's session
+                // req.body.session.jwt = token;
+                user.password = undefined;
+                res.cookie("token", token, {
+                    httpOnly: true,
+                    //secure: true, // only works on https    
+                });
+                console.log("r80", req.header);
```
Comment (Nirodie) [inline cors-server/routes/authentication.ts]: Testing console log should be removed

File: src/components/ChatRouter.tsx [hunk 10]
```diff
@@ -2,7 +2,6 @@ import { BrowserRouter as Router, Routes, Route} from "react-router-dom"
 import App from "../App";
 import Login from "./Login";
 import Register from "./Register"
-import Terms from "./Terms";
```
Comment (Nirodie) [inline src/components/ChatRouter.tsx:5]: Merge dev into the branch

**Review by Nirodie — CHANGES_REQUESTED** [review]
Remove console logs and merge dev into branch

**Review by M0ller — APPROVED** [review]

**Review by qoentz — APPROVED** [review]

**Pull request merged**